test/migration/setup_runner.go

```diff
@@ -91,6 +98,44 @@ func (r *SetupMigrationRunner) createAndWaitForSpace(t *testing.T, name, tierNam
 	}
 }
 
+func (r *SetupMigrationRunner) prepareProvisionedSubspace(t *testing.T) {
+	hostAwait := r.Awaitilities.Host()
+	memberAwait := r.Awaitilities.Member2()
+	r.createAndWaitForSpace(t, ProvisionedParentSpace, "base", memberAwait)
+	memberCluster, found, err := hostAwait.GetToolchainCluster(t, cluster.Member, memberAwait.Namespace, nil)
+	require.NoError(t, err)
+	require.True(t, found)
+
+	srClusterRoles := []string{commoncluster.RoleLabel(commoncluster.Tenant)}
+	t.Logf("creating space request %v for parent space %v", ProvisionedSpaceRequest, ProvisionedParentSpace)
+	spaceRequest := tsspace.CreateSpaceRequestForParentSpace(t,
+		r.Awaitilities,
+		memberAwait.ClusterName,
+		ProvisionedParentSpace,
+		tsspace.WithName(ProvisionedSpaceRequest),
+		tsspace.WithSpecTargetClusterRoles(srClusterRoles),
+		tsspace.WithSpecTierName("base"))
+
+	subSpace, err := hostAwait.WaitForSubSpace(t,
+		spaceRequest.GetName(),
+		spaceRequest.GetNamespace(),
+		ProvisionedParentSpace,
+		wait.UntilSpaceHasConditions(wait.Provisioned()),
+		wait.UntilSpaceHasAnyProvisionedNamespaces())
+	require.NoError(t, err)
+
+	_, err = memberAwait.WaitForSpaceRequest(t,
+		types.NamespacedName{
+			Namespace: spaceRequest.Namespace,
+			Name:      spaceRequest.Name,
+		},
+		wait.UntilSpaceRequestHasStatusTargetClusterURL(memberCluster.Spec.APIEndpoint),
+		wait.UntilSpaceRequestHasNamespaceAccess(subSpace),
+		wait.UntilSpaceRequestHasConditions(wait.Provisioned()),
+	)
```

Comment:
you could significantly simplify it to decrease the dependency on the actual implementation

Let's assume that whatever is in master just works, we don't need to verify all the details, just wait until the SpaceRequest is completed. The actual verification should be executed when the operator is updated with the changed version.

```suggestion
	_, err = memberAwait.WaitForSpaceRequest(t,
		types.NamespacedName{
			Namespace: spaceRequest.Namespace,
			Name:      spaceRequest.Name,
		},
		wait.UntilSpaceRequestHasConditions(wait.Provisioned()),
	)
```

Reply:
Fixed in 30ff3cf.